client/src/context/user.js

```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const UserContext = React.createContext([{}, () => {}]);
+
+export const UserProvider = props => {
+  const [user, setUser] = React.useState([{}]);
```

Comment:
Any reason why you have an array here?
`user` sounds like an object to me.

```suggestion
  const [user, setUser] = React.useState({});
```